### ISSUE TITLE
Set Autocomplete output file path

### ISSUE THREAD

**Issue by hongkongkiwi** [earthly/earthly#3067]
**Use case**

When using earthly `boostrap --with-autocomplete` the default zsh path used is `/usr/share/zsh/site-functions/_earthly`, however this path requires root access.

I would like the option to set a path to save the autocomplete files to, in my case I'm using `$HOME/.zfunc/_earthly`.

It would be great to have two options to override the default autocomplete path's (one for bash and one for zsh).

Alternatively, it would be great to simply output the autocomplete content and I can save it to my own files, e.g.
`earthly autocomplete zsh > "$HOME/.zfunc/_earthly"`

**Expected Behavior**

earthly bootstrap --with-autocomplete --zsh-autocomplete-file="$HOME/.zfunc/_earthly" --bash-autocomplete-file="$HOME/bash-autocomplete/earthly"

**Comment by hongkongkiwi** [comment]
I guess technically this is also a bug since I can't install the autocomplete on Mac OSX at the moment for zsh due to this issue.

**Comment by alexcb** [comment]
> however this path is blocked access

I agree that it sounds like a bug; we should fix how earthly deteremines the installation path.

The ability to override it via a flag could be nice, but ideally (mac) users shouldn't have to know where to install it.

**Comment by hongkongkiwi** [comment]
Hmmm this is tricky on OSX and depends on the users setup:
1. User is using builtin zsh which is located at /bin/zsh
2. User is using homebrew zsh version which is located at /opt/homebrew/bin/zsh

We can test which one by using `which zsh` to check path.

So, in the case of builtin zsh the files in theory are stored in a zsh version specific folder e.g.  `/usr/share/zsh/$(/bin/zsh --version | cut -f2 -d ' ')/functions`. But you can't write to this dir because OSX considers it a protected system file (even with sudo). I'm not sure how to get custom completion without a custom path in this case.

In the case of homebrew zsh, the path should be at (I think): `$(brew --prefix)/share/zsh/site-functions`, many homebrew formulas just drop the completion file in here by default. Take rust for example:
```
==> rust
zsh completions have been installed to:
  /opt/homebrew/share/zsh/site-functions
```

So, I'm not sure it makes sense to hardcode a path for the completion, but rather do what many tools do which is have an option to output it to the screen then just > into the correct file (if installed via homebrew it should default to homebrews correct path).

**Comment by alexcb** [comment]
> output it to the screen

This already exists, but might be a bit hidden/undocumented:

```
$ earthly bootstrap --source zsh
#compdef _earthly earthly

function _earthly {
    autoload -Uz bashcompinit
    bashcompinit
    complete -o nospace -C '/home/alex/bin/earthly-v0.7.11' earthly
}
```

It was added specifically for homebrew:

```
earthly bootstrap --help 2>&1 | grep -e --source
   --source value          Output source file (for use in homebrew install)
```

but perhaps we just need to update the help usage text, and document it somewhere.

**Comment by eliottwiener-gridunity** [comment]
> ```
> $ earthly bootstrap --no-buildkit --source zsh
> #compdef _earthly earthly
> 
> function _earthly {
>     autoload -Uz bashcompinit
>     bashcompinit
>     complete -o nospace -C '/home/alex/bin/earthly-v0.7.11' earthly
> }
> ```

@alexcb Shouldn't the `compdef` line be uncommented for this to work?

**Comment by alexcb** [comment]
@eliottwiener-gridunity I'm not a zsh user; however according to https://github.com/zsh-users/zsh-completions/blob/master/zsh-completions-howto.org it's correct syntax.

**Comment by eliottwiener-gridunity** [comment]
I see. This outputs the contents of a compdef file, which is picked up by zsh to be in the `$fpath` when `compinit` is invoked. Thanks.
